### ISSUE TITLE
Snapshot preference make no difference?

### ISSUE THREAD

**Issue by sophy7074** [igvteam/igv#640]
Dear IGV developers,

I'm working to sanpshot variants using batch script.
But after set preference in *.bat, I got the pictures that nearly same as before.

```
new
genome hg19
snapshotDirectory igv
load 6.1.hg19.bam
load 2.2.hg19.bam

maxPanelHeight 1000

preference SAM.SHADE_BASE_QUALITY false
preference SAM.QUICK_CONSENSUS_MODE false
preference SAM.SHOW_ALL_BASES true
preference SAM.SHOW_SOFT_CLIPPED true
preference SAM.DOWNSAMPLE_READS false
preference SAM.SHADE_CENTER false

goto 4:190976495-190976495

snapshot 4_190976495_190976495_test.png
exit

```

![image](https://user-images.githubusercontent.com/8351253/54578555-24a79880-4a3b-11e9-83ac-c1e73ee2faca.png)

- before

![4_190976495_190976495_h500_test o](https://user-images.githubusercontent.com/8351253/54578450-d09cb400-4a3a-11e9-9806-e52a7652ccc6.png)

- after

![4_190976495_190976495_h500_test (2)](https://user-images.githubusercontent.com/8351253/54578474-e0b49380-4a3a-11e9-8965-a7834932b26a.png)

I want to get a picture which **show all bases**/**show mismatches**/**not shade base by quality**/**not quick consensus mode**.
How to achieve it?

Thank you,
Tiany

**Comment by jrobinso** [comment]
I'll look into it.    I am the "IGV Developers" BTW,  please be patient there is a lot on my plate.  In the meantime, as a workaround, try setting those preferences in the UI (Preferences window).  Those settings will persist when you run batch scripts, or until you change them.

I see insertion markers are on,  did you enable that on purpose?   We released that by accident but quickly corrected it.

**Comment by sophy7074** [comment]
UI setting is OK。
The reson why using snapshot is that I need lots of variants pictures. So I want snapshot variants atuomatic.

About insertion markder, I didn't setting it.

The command I run is:
`sh /export/home/tianyu/software/IGV_Linux_snapshot/igv.sh -b igv/test.bat`

```
#!/bin/sh

#This script is intended for launch on *nix machines

#-Xmx4g indicates 4 gb of memory, adjust number up or down as needed
#Script must be in the same directory as igv.jar
#Add the flag -Ddevelopment = true to use features still in development
#Add the flag -Dsun.java2d.uiScale=2 for HiDPI displays
prefix=`dirname $(readlink $0 || echo $0)`

# Check whether or not to use the bundled JDK
if [ -d "${prefix}/jdk-11" ]; then
    echo echo "Using bundled JDK."
    JAVA_HOME="${prefix}/jdk-11"
    PATH=$JAVA_HOME/bin:$PATH
else
    echo "Bundled JDK not found.  Using system JDK."
fi

exec java --module-path="${prefix}/lib" -Xmx4g \
    @"${prefix}/igv.args" \
    -Dapple.laf.useScreenMenuBar=true \
    -Djava.net.preferIPv4Stack=true \
    --module=org.igv/org.broad.igv.ui.Main "$@"
```

**Comment by jrobinso** [comment]
OK, yes if you use the "snapshot" you will see development features, like
the insertion markers.

**Comment by jrobinso** [comment]
Apologies for the delay.  I just pushed a fix for this, it required a code change.  The fix will be in the next release (2.7.1) .   Even with the fix you need to set your preferences before loading the bam files, here is the script I used for testing

```
new
genome hg38

preference SAM.SHADE_BASE_QUALITY false
preference SAM.QUICK_CONSENSUS_MODE false
preference SAM.SHOW_ALL_BASES false
preference SAM.SHOW_SOFT_CLIPPED true
preference SAM.DOWNSAMPLE_READS false
preference SAM.SHADE_CENTER false

load https://s3.amazonaws.com/igv.org.test/data/pacbio/hg38_inv.bam
goto chr21:26016892-26020218
maxPanelHeight 1000

snapshot test.png
```

**Comment by sophy7074** [comment]
Cool, I will try it when the next version is available.
